src/protocol_common.c

```diff
@@ -295,7 +295,7 @@ esp_loader_error_t loader_spi_parameters(uint32_t total_size)
     return send_cmd(&spi_cmd, sizeof(spi_cmd), NULL);
 }
 
-__attribute__ ((weak)) void loader_port_debug_print(const char *str)
```

Comment:
Probably leftover code from testing, a weak symbol shouldn't hurt anything.

Reply:
@DNedic I think the reason is that `__attribute__((weak))` syntax is a GNU extension, and is not supported by MSVC compiler. Same for `packed` — MSCV has a different way of specifying the same.

I think this will still need to be cleaned up to move all the compiler-specific things into a common header.

Reply:
I see, I assumed MSVC supported GNU extensions wrongly, yes, we should move these platform details into a separate header if we merge this.

Reply:
Since LLVM supports these, is there a way to build for windows using LLVM and just not bother with MSVC? As far as I've seen the incompatibilities go deeper than GNU extensions.

Reply:
Mingw might work, but I'd prefer an "if msvc else" precompiler flag. Is does actually compile but the problem is this symbol name is being used multiple times and thus the linker complains. Renaming one of them would also fix the issue.